### PR TITLE
tests: adhere to standard

### PR DESCRIPTION
**Pull request by phisco**
### Description of your changes

Follow up to https://github.com/crossplane/crossplane/pull/4768/files/ac3f3cc72726c40ba956434013f3734a7bdf14b3#r1360377649.

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Added or updated unit **and** E2E tests for my change.
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] ~Added `backport release-x.y` labels to auto-backport this PR, if necessary.~
- [ ] ~Opened a PR updating the [docs], if necessary.~

[contribution process]: https://git.io/fj2m9
[docs]: https://docs.crossplane.io/contribute/contribute

### CODE REVIEW

**Review by pedjak — APPROVED** [review]
nit: the word  `properly` feels redundant in the added reasons.

File: internal/xcrd/crd_test.go [hunk 1]
```diff
@@ -1755,13 +1755,13 @@ func TestSetCrdMetadata(t *testing.T) {
...
-		want *extv1.CustomResourceDefinition
+	tests := map[string]struct {
+		reason string
+		args   args
+		want   *extv1.CustomResourceDefinition
 	}{
-		{
-			name: "set crd annotations",
+		"SetAnnotations": {
+			reason: "Should properly set CRD annotations only from XRD spec",
```
Comment (phisco) [inline internal/xcrd/crd_test.go:1764]:
```suggestion
			reason: "Should set CRD annotations only from XRD spec",
```

File: internal/xcrd/crd_test.go [hunk 2]
```diff
@@ -1791,8 +1791,8 @@ func TestSetCrdMetadata(t *testing.T) {
 				},
 			},
 		},
-		{
-			name: "set crd labels",
+		"SetLabelsFromXRDSpec": {
+			reason: "Should properly set CRD labels from XRD spec",
```
Comment (phisco) [inline internal/xcrd/crd_test.go:1795]:
```suggestion
			reason: "Should set CRD labels from XRD spec",
```

File: internal/xcrd/crd_test.go [hunk 3]
```diff
@@ -1821,8 +1821,8 @@ func TestSetCrdMetadata(t *testing.T) {
 				},
 			},
 		},
-		{
-			name: "append labels",
+		"AppendLabelsFromXRDSpec": {
+			reason: "Should properly set CRD labels by appending labels from the XRD spec to the ones of the XRD itself",
```
Comment (phisco) [inline internal/xcrd/crd_test.go:1825]:
```suggestion
			reason: "Should set CRD labels by appending labels from the XRD spec to the ones of the XRD itself",
```

File: internal/xcrd/crd_test.go [hunk 4]
```diff
@@ -1857,8 +1857,8 @@ func TestSetCrdMetadata(t *testing.T) {
 				},
 			},
 		},
-		{
-			name: "labels and annotations",
+		"SetLabelsAndAnnotations": {
+			reason: "Should properly set CRD labels and annotations from XRD spec and XRD itself",
```
Comment (phisco) [inline internal/xcrd/crd_test.go:1861]:
```suggestion
			reason: "Should set CRD labels and annotations from XRD spec and XRD itself",
```

**Pull request merged**